### PR TITLE
[PM-8517] Create web specific layout

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

We currently duplicate some logic between our layouts. In an effort to streamline our experience I'm exploring if we can create a web specific layout that handles some of this.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html:1]: SM for some reason didn't have `PaymentMethodWarningBanners`

File: apps/web/src/app/layouts/web-layout.component.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { ConfigService } from "@bitwarden/common/platform/abstractions/config/config.service";
+import { LayoutComponent, LayoutVariant } from "@bitwarden/components";
+
+import { PaymentMethodWarningsModule } from "../billing/shared";
+
+import { ProductSwitcherModule } from "./product-switcher/product-switcher.module";
+import { ToggleWidthComponent } from "./toggle-width.component";
+
+@Component({
+  selector: "app-layout",
```
Comment (willmartian) [inline apps/web/src/app/layouts/web-layout.component.ts:14]: Thoughts on `selector: "web-layout"`?

**Review by willmartian — DISMISSED** [review]
Nice. A lot less redundancy. One question:

File: bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html
Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html:1]: cc @jtouchstonebw @amorask-bitwarden do you know if this was intentional?

File: apps/web/src/app/layouts/web-layout.component.ts (hunk 1 above)
Comment (Hinton) [reply apps/web/src/app/layouts/web-layout.component.ts:14]: I was debating back and forth, we generally use `app` but I agree web would be nicer ...

File: bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html
Comment (jtouchstonebw) [reply bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html:1]: Not to my knowledge.

Comment (Hinton) [reply bitwarden_license/bit-web/src/app/secrets-manager/layout/layout.component.html:1]: 👍, they should all have the banner now.

**Review by Thomas-Avery — DISMISSED** [review]
SM changes look good.

**Review by addisonbeck — DISMISSED** [review]

**Review by Thomas-Avery — DISMISSED** [review]
SM changes look good.

**Review by addisonbeck — DISMISSED** [review]

File: apps/web/src/app/layouts/web-layout.component.ts (hunk 1 above)
Comment (vleague2) [reply apps/web/src/app/layouts/web-layout.component.ts:14]: Just to clarify for myself -- is the plan to change the selector to `web-layout` to match the file name? Or are we keeping `app-layout`?

File: libs/components/src/navigation/index.ts [hunk 2]
```diff
@@ -1,2 +1,3 @@
 export * from "./navigation.module";
 export * from "./side-nav.service";
+export type * from "./side-nav.component";
```
Comment (willmartian) [inline libs/components/src/navigation/index.ts:3]: Ah cool, I didn't know you could do `export type *`.

**Review by willmartian — APPROVED** [review]
LGTM!

File: apps/web/src/app/layouts/web-layout.component.ts (hunk 1 above)
Comment (willmartian) [reply apps/web/src/app/layouts/web-layout.component.ts:14]: I am personally ok with either.

**Review by willmartian — DISMISSED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by willmartian — APPROVED** [review]

**Pull request merged**